test/e2e/operator_test.go

```diff
@@ -350,3 +355,129 @@ func customResolver(nameserver string) *net.Resolver {
+	testIngressNamespace := "test-extdns-openshift-route"
+	t.Log("Ensuring test namespace")
+	err := kubeClient.Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testIngressNamespace}})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to ensure namespace %s: %v", testIngressNamespace, err)
+	}
+	openshiftRouterName := "external-dns"
+	name := types.NamespacedName{Namespace: testIngressNamespace, Name: openshiftRouterName}
+	t.Logf("Create custome ingress controller %s/%s", name.Namespace, name.Name)
+	ing := newHostNetworkController(name, name.Name+"."+hostedZoneDomain)
+	if err = kubeClient.Create(context.TODO(), ing); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer kubeClient.Delete(context.TODO(), ing)
+
+	externalDnsServiceName := fmt.Sprintf("%s-source-as-openshift-route", testExtDNSName)
+	t.Logf("Creating external dns instance : %s", externalDnsServiceName)
+	extDNS := helper.buildOpenShiftExternalDNS(externalDnsServiceName, hostedZoneID, hostedZoneDomain, openshiftRouterName)
+	if err = kubeClient.Create(context.TODO(), &extDNS); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to create external DNS %q: %v", testExtDNSName, err)
+	}
+	defer kubeClient.Delete(context.TODO(), &extDNS)
+
+	// Create conflicting routes in the namespaces
+	makeRoute := func(name types.NamespacedName, host, path string) *routev1.Route {
+		return &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: name.Namespace,
+				Name:      name.Name,
+				Annotations: map[string]string{
+					"external-dns.mydomain.org/publish": "yes",
+				},
+				Labels: map[string]string{
+					"external-dns": "",
+				},
+			},
+			Spec: routev1.RouteSpec{
+				Host: host,
+				Path: path,
+				To: routev1.RouteTargetReference{
+					Kind: "Service",
+					Name: "testServiceName",
+				},
+			},
+		}
+	}
+	// use unique names for each test route to simplify debugging.
+	routeName := types.NamespacedName{Namespace: testIngressNamespace, Name: "external-dns-route"}
+	route := makeRoute(routeName, "app."+hostedZoneDomain, "/apis")
+
+	// The first route should be admitted
+	if err = kubeClient.Create(context.TODO(), route); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to create route: %v", err)
+	}
+	defer kubeClient.Delete(context.TODO(), route)
+	canonicalName := ""
+	t.Logf("Getting canonicalName for the route :%s ", routeName.Name)
+	if canonicalName, err = fetchRouterCanonicalHostname(t, routeName); err != nil {
+		t.Fatalf("Failed to get RouterCanonicalHostname for route %s/%s: %v", routeName.Namespace, routeName.Name, err)
+	}
+	t.Logf("canonicalName  : %s for the route :%s ", routeName.Name, canonicalName)
+	verifyOpenShiftRouteSource(t, canonicalName, fmt.Sprintf("app.%s", hostedZoneDomain))
+}
+
+func verifyOpenShiftRouteSource(t *testing.T, canonicalName, host string) {
```

Comment:
What this function is verifying is that the given host is resolved to the given CNAME, not the source of type `Route` in ExternalDNS custom resource. Something like `verifyCNAMERecord` would better reflect the contents.
BTW do we really need this function? Why not adding its body directly to the test code?

Reply:
name Change done
copying directly is not ok, it looks clumsy  :( 
i prefer more granular(which will speaks more about the tasks like setup , actions, verification ...) but it will leads to lot more changes.

Reply:
Then it can be used in `TestExternalDNSWithRoute` test case too. But fine, let's keep it for the followup PRs.